### PR TITLE
#168402742 Validations

### PR DESCRIPTION
**Pull request by william000000**
#### What does this PR do?

-  Validate my routes, methods, and params

#### Description of the task to be completed

- Validate methods
- Validate routes
- Test them
- Validate params
- Add message in the response

#### How should this be manually tested?
- clone my repo `https://github.com/william000000/Free-Mentors.git`
- Install all dependencies by running  `npm i` 
- Then open vs code terminal, write `npm run dev` to start my server
- Then open postman and use `localhost:9000/api/v2/auth/signin` as` POST` method and try when method is `DELETE`
#### What are relevant pivotal tracker stories
[Finishes #168402742](https://www.pivotaltracker.com/story/show/168402742)
#### Any background context you want to provide?
- N/A

#### Screenshot

image

image
image

#### Questions:
- N/A

### CODE REVIEW

File: Server/src/app.js [hunk 1]
```diff
@@ -7,11 +7,12 @@ import sessionRoute from './routes/mentorship.route';
 import reviewRoute from './routes/review.routes';
 import swaggerUI from 'swagger-ui-express';
 import mySwagger from '../../swagger.json';
+import http from 'http';
+const s = http.createServer();
```
Comment (Kasulejoseph) [inline Server/src/app.js]: `S` is not a good variable name, I can't even see where you using it

File: Server/src/routes/admin.routes.js [hunk 2]
```diff
@@ -7,4 +7,17 @@ const router = express.Router();
 const { validatePath } = validateRoute;
 router.patch('/:userId', checkIsAdmin, validatePath, adminController.changeUserToMentor);
 
+router.get('/:userId', (req, res) => {
+  return res.status(405).json({ status: 405, error: 'Incorrect Method' });
+});
+router.post('/:userId', (req, res) => {
+  return res.status(405).json({ status: 405, error: 'Incorrect Method' });
```
Comment (Kasulejoseph) [inline Server/src/routes/admin.routes.js]: You need to get a better way of implementing this, Imagine you had 100 routes would you go doing this?

File: Server/src/routes/admin.routes.js [hunk 3]
```diff
@@ -5,6 +5,13 @@ import validateRoute from '../helpers/validateRoutes';
 
 const router = express.Router();
 const { validatePath } = validateRoute;
-router.patch('/:userId', checkIsAdmin, validatePath, adminController.changeUserToMentor);
+router.route('/:userId')
+  .patch(checkIsAdmin, validatePath, adminController.changeUserToMentor)
+  .all(methodNotCorrect);
+
+function methodNotCorrect(req, res) {
```
Comment (Kasulejoseph) [inline Server/src/routes/admin.routes.js]: Use ES6 functions

File: Server/src/controllers/mentors.controller.js [hunk 4]
```diff
@@ -15,7 +15,7 @@ class MentorController {
       const { isadmin, ...data } = findMentors[0];
       return res.status(200).json({ status: 200, data });
     } catch (err) {
-      return res.status(400).json({status: 400, error: err.message});
+      return res.status(400).json({status: 400, message: 'mentors retrieved succefully', error: err.message});
```
Comment (Kasulejoseph) [inline Server/src/controllers/mentors.controller.js]: You don't need to return a message on error. status and error are just enough

Comment (william000000) [reply Server/src/controllers/mentors.controller.js]: Ohh yeah I going to correct it as fast as possible

**Review by Kasulejoseph — APPROVED** [review]
Looks good to merge.

**Pull request merged**